mava/wrappers/gym.py

```diff
@@ -0,0 +1,279 @@
+    ) -> Tuple[NDArray, Dict]:
+        agents_view, info = self._env.reset(seed, options)
+
+        # Create the metrics dict
+        metrics = {
+            "episode_return": self.running_count_episode_return,
+            "episode_length": self.running_count_episode_length,
+            "is_terminal_step": True,
+        }
+
+        # Reset the metrics
+        self.running_count_episode_return = 0.0
+        self.running_count_episode_length = 0
+
+        if "won_episode" in info:
+            metrics["won_episode"] = info["won_episode"]
+
+        info["metrics"] = metrics
+
+        return agents_view, info
+
+    def step(self, actions: NDArray) -> Tuple[NDArray, NDArray, NDArray, NDArray, Dict]:
+        agents_view, reward, terminated, truncated, info = self._env.step(actions)
+
+        self.running_count_episode_return += float(np.mean(reward))
+        self.running_count_episode_length += 1
+
+        metrics = {
+            "episode_return": self.running_count_episode_return,
+            "episode_length": self.running_count_episode_length,
+            "is_terminal_step": False,
+        }
+        if "won_episode" in info:
+            metrics["won_episode"] = info["won_episode"]
+
+        info["metrics"] = metrics
+
+        return agents_view, reward, terminated, truncated, info
+
+
+class GymAgentIDWrapper(gymnasium.Wrapper):
+    """Add one hot agent IDs to observation."""
+
+    def __init__(self, env: gymnasium.Env):
+        super().__init__(env)
+
+        self.agent_ids = np.eye(self.env.num_agents)
+        self.observation_space = self.modify_space(self.env.observation_space)
+
+    def reset(
+        self, seed: Optional[int] = None, options: Optional[dict] = None
+    ) -> Tuple[NDArray, Dict]:
+        """Reset the environment."""
+        obs, info = self.env.reset(seed, options)
+        obs = np.concatenate([self.agent_ids, obs], axis=1)
+        return obs, info
+
+    def step(self, action: list) -> Tuple[NDArray, float, bool, bool, Dict]:
+        """Step the environment."""
+        obs, reward, terminated, truncated, info = self.env.step(action)
+        obs = np.concatenate([self.agent_ids, obs], axis=1)
+        return obs, reward, terminated, truncated, info
+
+    def modify_space(self, space: gymnasium.spaces) -> gymnasium.spaces:
+        if isinstance(space, gymnasium.spaces.Box):
+            new_shape = space.shape[0] + len(self.agent_ids)
+            return gymnasium.spaces.Box(
+                low=space.low, high=space.high, shape=new_shape, dtype=space.dtype
+            )
+        elif isinstance(space, gymnasium.spaces.Tuple):
+            return gymnasium.spaces.Tuple(self.modify_space(s) for s in space)
+        else:
+            raise ValueError(f"Space {type(space)} is not currently supported.")
+
+
+# Copied form Gymnasium/blob/main/gymnasium/vector/async_vector_env.py
+# Modified to work with multiple agents
+def async_multiagent_worker(  # noqa CCR001
```

Comment:
Are we keeping this in this PR and changing it later? If so can you just add a comment onto the lines that needed changing

Reply:
The only difference is making the autotest able to handle arrays, I added a comment on top of the modified line.